### PR TITLE
[RPRBLND-1657] Fixed RIF Transparent Background filter on OSX

### PR DESCRIPTION
**Pull request by ardraeiss**
### PURPOSE
OSX handles the RIF filter output buffer differently than Windows or Linux causing SEG_FAULT crash on transparent background filter application.
This PR fixes this behavior by switching filter to default input and output buffer names(“inputImage” and “outputImage”) instead of ImageFilterSetParameterImage usage.

### EFFECT OF CHANGE
- fixed Transparent Background crash in Full render quality;
- enabled Transparent Background for High, Medium and Low render qualities.

### TECHNICAL STEPS
- renamed input and output RIF filter buffers so Metal will handle filter application correctly;
- enabled RIF-based transparent background filter for Hybrid modes;
- added support for more useful argument types - 2-3-4-length tuples of unsigned integers or floats.

### NOTES
- RIF documentation for USER_DEFINED filter type should be adjusted to use correct input and output buffer names.

### CODE REVIEW

File: src/bindings/pyrpr/src/pyrprimagefilters.py [hunk 1]
```diff
@@ -305,9 +305,25 @@ def set_parameter(self, name, value):
         elif isinstance(value, Image):
             ImageFilterSetParameterImage(self, pyrpr.encode(name), value)
             self.parameters[name] = value
-        elif isinstance(value, tuple) and len(value) == 2:
+        elif isinstance(value, tuple) and len(value) > 0:
```
Comment (bsavery) [inline src/bindings/pyrpr/src/pyrprimagefilters.py:308]: please call len(value) once only.

File: src/rprblender/engine/render_engine.py [hunk 2]
```diff
@@ -476,7 +476,7 @@ def sync(self, depsgraph):
         self.setup_image_filter(image_filter_settings)
 
         # Shadow catcher
-        if scene.rpr.render_quality == 'FULL2':
+        if scene.rpr.render_quality != 'FULL':
```
Comment (bsavery) [inline src/rprblender/engine/render_engine.py:479]: so this should work for hybrid too?

Comment (ardraeiss) [reply src/rprblender/engine/render_engine.py:479]: it uses RIF only so it works on Hybrid

File: src/rprblender/engine/image_filter.py [hunk 3]
```diff
@@ -112,28 +112,26 @@ def get_data(self):
...
+    def setup_alpha_filter(self, alpha):
         """ Apply transparent background by setting output image alpha by alpha value """
         result = self.context.create_filter(rif.IMAGE_FILTER_USER_DEFINED)
 
         # redefine image alpha channel by alpha value
         code = \
             "int2 coord;" \
-            "GET_COORD_OR_RETURN(coord, GET_BUFFER_SIZE(outBuf));" \
-            "vec4 pixel = ReadPixelTyped(inputBuf, coord.x, coord.y);" \
+            "GET_COORD_OR_RETURN(coord, GET_BUFFER_SIZE(outputImage));" \
```
Comment (bnagirniak) [inline src/rprblender/engine/image_filter.py:122]: Looks like using 3 quotes syntax for multiline string would suit better here:
code = """
.....
"""

**Review by bnagirniak — COMMENTED** [review]
Left comment about code improvement.

File: src/rprblender/engine/image_filter.py (hunk 3 above)
Comment (bsavery) [reply src/rprblender/engine/image_filter.py:122]: +1

Comment (ardraeiss) [reply src/rprblender/engine/image_filter.py]: Yeah, that looks much better

**Review by bnagirniak — APPROVED** [review]
Code is ok.

**Pull request merged**